MozillaFoundation/foundation.mozilla.org
# 10846 make donate modal url editable

# Description




**Note**: This PR depends on #10892, please merge first and then take a look at this one.

Link to sample test page:
Related PRs/issues: #10846

This PR updates the model for the donation modal which appears after petition submission with a new field `donate_url`. 

This `donate_url` field allows users to enter a custom URL (Ex: http:// or https://) **or** relative URL (Ex: ?form=donate) if desired. 

If left blank, the donation modal will link to the same default donation form as before ("?form=donate").

This PR also adds tests for the donation modal's URL validation via the file `test_donation_modal.py`


# Steps to test

1. Check this branch out
2. Run `inv copy-prod-db`
3. Visit http://localhost:8000/cms/snippets/wagtailpages/donationmodal/edit/1/
4. Scroll down to "Donate Url" , enter an invalid URL (Ex: "test"), click "Publish", and note that you get a validation error.
5. Enter a valid URL, and click "Publish".
7. Visit [this campaign page](http://localhost:8000/en/campaigns/tell-slack-make-dms-safe/?thank_you=true) and click the purple "Yes, I'll chip in" button
8. If you are linked to the expected URL, testing is complete! 


# Screenshots

## Full URL 
### CMS
![Screenshot 2023-08-02 at 15-49-43 Editing donation modal - privacy_donate_modal - Wagtail](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/5cb14f9e-2bb9-48f0-9bd3-a4e129c8cf5a)

### Button Link
<img width="1912" alt="Screenshot 2023-08-02 at 3 50 11 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/34c417fa-eec0-4712-ae2c-246d6e59e687">


## Relative URL 
### CMS
![Screenshot 2023-08-02 at 15-52-17 Editing donation modal - privacy_donate_modal - Wagtail](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/4afbcedb-66e8-498c-9b33-6ce131d4d7fd)

### Button Link
<img width="1912" alt="Screenshot 2023-08-02 at 3 53 01 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/356421d5-d5f0-4f8a-9356-64440bf2e14d">


## Full URL with Params

### CMS
![Screenshot 2023-08-02 at 15-53-22 Editing donation modal - privacy_donate_modal - Wagtail](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/5069d012-9487-4265-854c-58fbf2b5b426)

### Button Link
<img width="1912" alt="Screenshot 2023-08-02 at 3 53 52 PM" src="https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/17f91c10-aa43-41be-a1a8-bcf6078ebe46">


## Invalid URL (Neither Full URL or Relative)
![Screenshot 2023-08-02 at 16-00-12 Editing donation modal - privacy_donate_modal - Wagtail](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/013e5710-f674-4b94-9214-ab991db6d65c)
